universal.md

```diff
@@ -0,0 +1,423 @@
+
+- **`objects`** *array of objects*
+
+  An OPTIONAL list of objects that make up this manifest. These will potentially include multiple objects of the same or differetn versions that the client will pick. Blob and reference links are extracted from this list for the server housekeeping activity. Note we could use a map rather than a list here, with keys corresponding to object `type`.
+
+The `objects` each correspond to items that the client may wish to process, and are defined as follows.
+
+- **`type`** *string*
+
+  A REQUIRED type for this object. Official allocations will be registered with OCI, such as for container images (eg `org.oci.image`), and other allocations will be recommended to be registered as standard. These should be reverse DNS allocations to avoid conflicts.
+
+- **`version`** *string*
+
+  A REQUIRED version string for the type. Clients will normally look for an object with the latest version they understand, and fall back to older versions if they support them. Different versions may have different structures.
+
+- **`filters`** *map string - string*
+
+  An array of OPTIONAL keys and values for selecting among multiple object options that a client has. These are filters that the client will use to select which version of an image is most appropriate, for example selecting by architecture ro any other appropriate choice. Common keys and values should be used where possible, such as `org.oci.architecture` so that common code and documentation can be reused. These are collected into a single map so a user interface can display them without understanding the object type.
+
+- **`components`** *array of components*
+
+  An array of OPTIONAL components of the object, such as layers and config for an image. See below for details, this is parsed to extract the `blob` and `reference` links for gc.
+
+- **`annotations`** *string-string map*
+
+  Standard OCI annotations as per existing standards, this is metadata that may have common fields across different objects. Arguably this should be a map of string to string array so that annotation keys can be repeated, so they map better to other structures (such as http headers) that allow repetition.
+
+- **any other data** *unspecified*
+
+  Any other fields with any other structure. The client can use this data in any way that is appropriate, but generic tooling will not usually be able to do anything with it as there are no standards.
+
+The `components` of an object correspond to individual parts, potentially of multiple types, for example an image has a config and some layers. The parts that are needed to extract generic data are specified but again any type specific data can be added.
+
+- **`rtype`** *string*
```

Comment:
I like the idea of a "reference type", and that's something I'd be interested in adding to OCI Content Descriptors.

The terminology here is a bit confusing, you have `rtype` as the "reference type" which can be either be "blob" or "reference". I got stuck here. Would "reference" imply this points to a non-opaque thing, e.g. another `application/vnd.oci.object.manifest.v1+json`? What about a manifest? Anything else?

Some diagrams would help me a lot here.

I'd be interested in an example that doesn't include `rtype`. This would be for things outside the DAG?